packages/ember-metal/lib/streams/key-stream.js

```diff
@@ -25,6 +25,12 @@ let KeyStream = BasicStream.extend({
 
   compute() {
     var object = this.sourceDep.getValue();
+    const type = typeof object;
+
+    if (type === 'boolean' || type === 'string') {
```

Comment:
So I moved things to `get` and placed the guards there. I will agree the string special case is gross.